src/matrices.jl

```diff
@@ -0,0 +1,76 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    JColumn{T}(d)
+
+Column of ones.
+"""
+struct JColumn{T} <: AbstractVector{T}
+  d::Int
+end
+JColumn(d::Int) = JColumn{Bool}(d)
+Base.size(A::JColumn) = (A.d,)
+Base.IndexStyle(::Type{<:JColumn}) = IndexLinear()
+Base.getindex(A::JColumn{T}, i::Integer) where {T} = one(T)
+
+
+"""
+    JMatrix{T}(d)
+
+Square matrix of ones.
+"""
+struct JMatrix{T} <: AbstractMatrix{T}
+  d::Int
+end
+JMatrix(d::Int) = JMatrix{Bool}(d)
+Base.size(A::JMatrix) = (A.d, A.d)
+Base.IndexStyle(::Type{<:JMatrix}) = IndexCartesian()
+Base.getindex(A::JMatrix{T}, i::Integer, j::Integer) where {T} = one(T)
+
+
+"""
+    FMatrix{T}(d)
+
+`F` matrix, as defined by Aitchison 1986.
+"""
+struct FMatrix{T} <: AbstractMatrix{T}
+  d::Int
+end
+FMatrix(d::Int) = FMatrix{Int}(d)
+F(d::Int) = FMatrix(d)
+Base.size(A::FMatrix) = (A.d, A.d+1)
```

Comment:
```suggestion
Base.size(F::FMatrix) = (F.d, F.d + 1)
```

Reply:
I was unsure about how Julia would handle conflicts of names, if the arguments of the functions always overwrite the other variables w/ that name I agree that using F looks much more readable.